src/scss/_custom.scss

```diff
@@ -0,0 +1,132 @@
+/// Transforms a custom theme map into brand variables. See `_brand.scss`.
+/// @access private
+/// @param {map|null} $theme - A custom theme map.
+/// @return {map} Brand variables.
+@function _oFormsThemeToBrandVariants($theme) {
+	@if $theme {
+			@if not index(map-keys($theme), 'controls-base') {
+				@error "The custom theme for box-styled controls is missing a 'controls-base' key. (see @mixin _oFormsAnchorBox)";
+			}
+		// Add default value for checked background if not specified.
+		@if not map-get($theme, 'negative-checked-background') {
+			$theme: map-merge($theme, (
+				'negative-checked-background': map-get($theme, 'checked-background')
+			));
+		}
+
+		$theme: (
+			controls-base: oColorsGetPaletteColor(map-get($theme, 'controls-base')),
+			controls-checked-base: oColorsGetPaletteColor(map-get($theme, 'controls-checked-base')),
+			controls-negative-checked-background: oColorsGetPaletteColor(map-get($theme, 'controls-negative-checked-background'))
+		);
+	}
+
+	@return $theme;
+}
+
+/// @access private
+/// Outputs base styling for custom icons
+@mixin _oFormsCustomIconBase {
+	label {
+		min-width: unset;
+	}
+
+	.o-forms__current,
+	input[type=radio]:checked + span {  // sass-lint:disable-line no-qualifying-elements
+		&:after {
+			display: block;
+		}
+	}
+}
+
+/// @access public
+/// @param {String} $icon-name Name of the icon to request from the image set
+/// @param {String} $icon-class Class name for custom icon
+/// @param {Map|null} $theme Custom theme map
+@mixin oFormsCustomIcon($icon-name, $icon-class, $theme: null) {
```

Comment:
Isn't this specif to the custom anchored radio button? Maybe `oFormsCustom` could accept `$opts` which could include a list of icons for the anchored radio button?